internal/subscription/core_test.go

```diff
@@ -193,3 +194,60 @@ func getSubModel() Model {
 	}
 	return sub
 }
+
+func TestCore_RescaleSubTopics(t *testing.T) {
+	type fields struct {
+		repo        IRepo
+		projectCore project.ICore
+		topicCore   topic.ICore
+	}
+	type args struct {
+		ctx        context.Context
+		topicModel *topic.Model
+		partitions int
+	}
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	mockProjectCore := pCore.NewMockICore(ctrl)
+	mockTopicCore := tCore.NewMockICore(ctrl)
+	mockRepo := repo.NewMockIRepo(ctrl)
+	topic := &topic.Model{
+		Name:          "topic",
+		NumPartitions: 2,
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Rescale without errors",
+			fields: fields{
+				repo: mockRepo,
+				projectCore: mockProjectCore,
+				topicCore: mockTopicCore,
+			},
+			args: args{
+				ctx: ctx,
+				topicModel: topic,
+				partitions: 2,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Core{
+				repo:        tt.fields.repo,
+				projectCore: tt.fields.projectCore,
+				topicCore:   tt.fields.topicCore,
+			}
+			mockRepo.EXPECT().List(gomock.Any(), gomock.Any())
+			if err := c.RescaleSubTopics(tt.args.ctx, tt.args.topicModel, tt.args.partitions); (err != nil) != tt.wantErr {
```

Comment:
What is being tested here? Can we verify that the kafka call to update partition is being called?

Reply:
added check on Update Topic call